app/services/gias/csv_transformer.rb

```diff
@@ -0,0 +1,55 @@
+module Gias
+  class CsvTransformer
+    include ServicePattern
+
+    def initialize(input_file)
+      @input_file = input_file
+      @output_file = Tempfile.new
+    end
+
+    def call
+      output_csv = CSV.new(output_file)
+
+      CSV.new(input_file, headers: true, return_headers: true).each do |row|
+        if row.header_row? || school_in_scope?(row)
+          output_csv << row
+        end
+      end
+
+      # Rewind the file so it's ready for reading
+      output_file.rewind
+      output_file
+    end
```

Comment:
I think you can wrap this into a block to remove the for manually rewinding the file.

```suggestion
    def call
      CSV.open(output_file, "w") do |output_csv|
        CSV.foreach(input_file, headers: true, return_headers: true) do |row|
          if row.header_row? || school_in_scope?(row)
            output_csv << row
          end
        end
      end
    end
```

Reply:
For the sake of consistency, the `Gias::CsvDownloader`, `Gias::CsvTransformer` and `Gias::CsvImporter` services all accept and/or return [File] objects.

Internally, `Gias::CsvTransformer` uses a [CSV] object to wrap the `output_file` (a [File] object). But this is an internal implementation detail – externally, the service still returns a regular [File] object.

I don't think [CSV.open] will work in this case because it expects a filename (a String) rather than an actual file handle. And it returns a [CSV] object, where we need to return a [File].

I don't think there's a CSV method that will accept a [File] and a block, and return the file rewound. I appreciate the code looks less elegant than your suggestion, but I think it's what we need to use in this case.

[File]: https://ruby-doc.org/3.2.2/File.html
[CSV]: https://ruby-doc.org/3.2.2/stdlibs/csv/CSV.html
[CSV.open]: https://ruby-doc.org/3.2.2/stdlibs/csv/CSV.html#method-c-open